linkcheck/logger/html.py

```diff
@@ -199,58 +210,98 @@ def write_parent(self, url_data):
             vhtml = validate_html % {'uri': url_data.parent_url}
             vcss = validate_css % {'uri': url_data.parent_url}
             self.writeln()
-            self.writeln('(<a href="'+vhtml+'">HTML</a>)')
-            self.write('(<a href="'+vcss+'">CSS</a>)')
+            self.writeln('(<a href="' + vhtml + '">HTML</a>)')
+            self.write('(<a href="' + vcss + '">CSS</a>)')
         self.writeln("</td></tr>")
 
     def write_base(self, url_data):
         """Write url_data.base_ref."""
-        self.writeln("<tr><td>"+self.part("base")+"</td><td>"+
-                     html.escape(url_data.base_ref)+"</td></tr>")
+        self.writeln(
+            "<tr><td>"
+            + self.part("base")
+            + "</td><td>"
+            + html.escape(url_data.base_ref)
+            + "</td></tr>"
+        )
 
     def write_real(self, url_data):
         """Write url_data.url."""
-        self.writeln("<tr><td>"+self.part("realurl")+"</td><td>"+
-                     '<a target="top" href="'+url_data.url+
-                     '">'+html.escape(url_data.url)+"</a></td></tr>")
+        self.writeln(
+            "<tr><td>"
+            + self.part("realurl")
+            + "</td><td>"
+            + '<a target="top" href="'
+            + url_data.url
```

Comment:
Note to self: create an issue to review this code and fix HTML injection problems.

Reply:
https://github.com/linkchecker/linkchecker/issues/439